networkx/algorithms/traversal/tests/test_bfs.py

```diff
@@ -51,6 +51,18 @@ def test_bfs_tree_isolates(self):
         assert sorted(T.nodes()) == [1]
         assert sorted(T.edges()) == []
 
+    def test_bfs_layers(self):
+        assert dict(enumerate(nx.bfs_layers(self.G, [0]))) == {
+            0: [0],
+            1: [1],
+            2: [2, 3],
+            3: [4],
+        }
+
+    def test_bfs_layers_missing_source(self):
+        with pytest.raises(nx.NetworkXError):
+            next(nx.bfs_layers(self.G, ["abc"]))
```

Comment:
Duplicate this test and the `pytest.raises` clause so you can test input `["abc"]` and then `"abc"`.